dask_geomodeling/raster/elemwise.py

```diff
@@ -746,3 +749,81 @@ def process(process_kwargs, *args):
             values[index] = data[index]
 
         return {"values": values, "no_data_value": fillvalue}
+
+
+class BaseLogExp(BaseSingle):
+    """
+    Base Block for Exp, Log and Log10 Blocks.
+    """
+    def __init__(self, x):
+        if x.dtype == np.dtype("bool"):
+            raise TypeError("input block must not have boolean dtype")
+        super(BaseLogExp, self).__init__(x)
+
+    def get_sources_and_requests(self, **request):
+        process_kwargs = {"dtype": self.dtype.name, "fillvalue": self.fillvalue}
+        return [(process_kwargs, None), (self.args[0], request)]
+
+    @property
+    def dtype(self):
+        # use at least float32
+        return np.result_type(np.float32, *self.args)
```

Comment:
Numpy uses at least np.float16

```
In [6]: np.log10(np.array([1], dtype=np.float16))                             
Out[6]: array([0.], dtype=float16)

In [7]: np.log10(np.array([1], dtype=np.float32))                             
Out[7]: array([0.], dtype=float32)

In [9]: np.log10(np.array([1], dtype=np.int8))                                
Out[9]: array([0.], dtype=float16)

In [10]: np.log10(np.array([1], dtype=np.int16))                              
Out[10]: array([0.], dtype=float32)
```

But that is a bit weird dtype. I am :+1: on changing this to float32 in dask-geomodeling.